### DISCUSSION TITLE
Issue Connecting JupyterHub to Jupyter Enterprise Gateway 2.6.0

### DISCUSSION THREAD

**Discussion by alex-g-tejada** [jupyter/enterprise_gateway#1082] [Q&A] [answered]
Hey everyone!

I was working with JEG with Jupyterhub in Kubernetes. Before I was using the dev image of the enterprise gateway and now I wanted to switch over to version 2.6. With the dev image, the enterprise gateway would be able to find all the kernels I defined in /usr/local/share/jupyter/kernel/

The new image for enterprise gateway is showing this error:
`Fetching kernelspecs from 'http://localhost:8888/api/kernelspecs' `
`kernel_image_puller.MainThread] Got exception attempting to retrieve kernelspecs - retrying. Exception was: HTTPConnectionPool(host='localhost', port=8888): Max retries exceeded with url: /api/kernelspecs (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f3122b7a860>: Failed to establish a new connection: [Errno 111] Connection refused'))`

The kernels were moved into the kernel folder but the pod is having a tough time locating them.

On the Jupyterhub side, I am authenticating this way:
`JUPYTER_GATEWAY_URL: "https://enterprise-gateway.my.domain.com"`
`JUPYTER_GATEWAY_HTTP_USER: guest`
`JUPYTER_GATEWAY_HTTP_PWD: guest-password`
`JUPYTER_GATEWAY_VALIDATE_CERT: "false"`

Does anyone have any idea why this might now be working?

**Comment by kevin-bates** [comment, marked as answer]
The kernel spec location will be the same in 2.6. This looks more like an issue with the deployment of EG itself.  The message you have posted appears more like it's from the `KernelImagePuller` pod (see: `kernel_image_puller.MainThread`) that runs on each node of the cluster (via a daemonset), and its unable to access the EG server to gather the kernelspecs so it can determine which images to pull.

I would first checked the EG pod logs for anything suspicious.  You might also try directing a `curl` command the the EG service
s `/api/kernelspecs` endpoint, similar to the following (with the appropriate host and port items adjusted of course)...
```bash
curl -X GET -i http://localhost:30443/api/kernelspecs
```
and confirm you're getting the set of kernelspecs from your EG service in the first place.

**Reply by alex-g-tejada** [reply]
Thanks Kevin! I think my images were mismatched
